lib/puppet/pops/types/types.rb

```diff
@@ -364,6 +402,16 @@ def eql?(o)
 # @api public
 #
 class PType < PTypeWithContainedType
+
+  def self.register_ptype(loader, ir)
+    create_ptype(loader, ir, 'AnyType',
+       'type' => {
+         KEY_TYPE => POptionalType.new(PType::DEFAULT),
+         KEY_VALUE => nil
+       }
+    )
+  end
```

Comment:
isn't this type abstract? Why have this here -seems repeated in derived classes (I looked at PNotUndefType).